Model/Sync/Checkout/Data.php

```diff
@@ -204,28 +218,37 @@ public function getDataBeforeChange()
     }
 
     /**
-     * Prepare address data
+     * Prepare billing address data
      * @param Quote $quote
-     * @param string $type
      * @return array<mixed>
      * @throws LocalizedException
      */
-    public function prepareAddress(Quote $quote, string $type)
+    public function prepareBillingAddress(Quote $quote)
     {
-        if ($type == 'billing') {
-            $address = $quote->getData('newBillingAddress') ?: $quote->getBillingAddress();
-            if (!$address->getCountryId() && $quote->getIsVirtual()) {
-                $customer = $quote->getCustomer();
-                /** @phpstan-ignore-next-line */
-                $customerAddressId = $customer->getDefaultBilling();
-                if ($customerAddressId) {
-                    $address = $this->customerAddressRepository->getById($customerAddressId);
-                }
+        $billingAddress = $quote->getData('newBillingAddress') ?: $quote->getBillingAddress();
+        if (!$billingAddress->getCountryId() && $quote->getIsVirtual()) {
+            $customer = $quote->getCustomer();
+            /** @phpstan-ignore-next-line */
+            $defaultCustomerBillingAddressId = $customer->getDefaultBilling();
+            if ($defaultCustomerBillingAddressId) {
+                $billingAddress = $this->customerAddressRepository->getById($defaultCustomerBillingAddressId);
             }
-        } else {
-            $address = $quote->getShippingAddress();
         }
-        return $this->abstractData->prepareAddressData($address);
+
+        return $this->abstractData->prepareAddressData($billingAddress);
```

Comment:
You deleted the fallback to prepareShippingAddress, which was used.

Reply:
I didn't, because its was not a fallback :), it is a if condition for 'billing' and 'shipping' perparation, so it was only getting there if it was indeed a shippingAddress.